Source/Evolve/WarpXEvolveEM.cpp

```diff
@@ -89,14 +90,23 @@ WarpX::EvolveEM (int numsteps)
         } else {
             // Beyond one step, we have E^{n} and B^{n}.
             // Particles have p^{n-1/2} and x^{n}.
-            FillBoundaryE();
-            FillBoundaryB();
-            UpdateAuxilaryData();
 
+            // E and B are up-to-date inside the domain only
+            FillBoundaryE(guard_cells.ng_FieldGather, guard_cells.ng_Extra);
+            FillBoundaryB(guard_cells.ng_FieldGather, guard_cells.ng_Extra);
+            // E and B: enough guard cells to update Aux or call Field Gather in fp and cp
+            // Need to update Aux on lower levels, to interpolate to higher levels.
+#ifndef WARPX_USE_PSATD
+            FillBoundaryAux(guard_cells.ng_UpdateAux);
+#endif
```

Comment:
I am not sure why this is needed for PSATD. Could you explain in which context you noticed that this was needed?

Reply:
This is an `#ifndef`, not an `#ifdef`.